src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/OngoingTasksHandlerProcessorForDeleteSubscriptionTasks.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.Handlers.Processors.OngoingTasks
+{
+    internal class OngoingTasksHandlerProcessorForDeleteSubscriptionTasks : AbstractOngoingTasksHandlerProcessorForDeleteSubscriptionTasks<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public OngoingTasksHandlerProcessorForDeleteSubscriptionTasks([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask DeleteOngoingTaskAsync()
+        {
+            using (var processor = new OngoingTasksHandlerProcessorForDeleteOngoingTask(RequestHandler))
```

Comment:
1. OngoingTasksHandlerProcessorForDeleteSubscriptionTasks processor should inherit from the OngoingTasksHandlerProcessorForDeleteOngoingTask processor.
2. Override ExecuteAsync
3. Do the Subscription Task validation
4. Execute base.ExecuteAsync();

Reply:
Same for the other one.

Reply:
done